### PR TITLE
Convert hui-toggle-entity-row to TypeScript/LitElement

### PR DESCRIPTION
**Pull request by iantrich**

### CODE REVIEW

File: src/panels/lovelace/entity-rows/hui-toggle-entity-row.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    this._config = config;
+  }
+
+  protected render(): TemplateResult {
+    if (!this.hass || !this._config) {
+      return html``;
+    }
+
+    const stateObj = this.hass.states[this._config.entity];
```
Comment (zsarnett) [inline src/panels/lovelace/entity-rows/hui-toggle-entity-row.ts]: Should we be checking if this exists? Or is this a case of to even call this element there has to be a state obj?

File: src/panels/lovelace/entity-rows/hui-toggle-entity-row.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            : html`
+            <div>
+              ${this._computeState(stateObj)}
+            </div>`
+        }
+      </hui-generic-entity-row>
+    `;
+  }
+
+  private _computeState(stateObj: HassEntity) {
```
Comment (zsarnett) [inline src/panels/lovelace/entity-rows/hui-toggle-entity-row.ts]: Return Type?

File: src/panels/lovelace/common/create-row-element.js [hunk 3]
```diff
@@ -12,7 +12,7 @@ import "../entity-rows/hui-scene-entity-row.js";
 import "../entity-rows/hui-script-entity-row.js";
 import "../entity-rows/hui-text-entity-row.js";
 import "../entity-rows/hui-timer-entity-row.js";
-import "../entity-rows/hui-toggle-entity-row.js";
+import "../entity-rows/hui-toggle-entity-row";
```
Comment (balloob) [inline src/panels/lovelace/common/create-row-element.js]: We should just have a PR that rips out all `.js` from all imports throughout the whole repo.

File: src/panels/lovelace/entity-rows/hui-toggle-entity-row.ts (hunk 1 above)
Comment (balloob) [reply src/panels/lovelace/entity-rows/hui-toggle-entity-row.ts]: We should check if stateObj is defined. Might want to consider updating the type of `hass.states` to indicate that we don't have a value for every key.

File: src/panels/lovelace/entity-rows/hui-toggle-entity-row.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    return stateObj && computeStateDisplay(this.localize, stateObj);
```
Comment (balloob) [inline src/panels/lovelace/entity-rows/hui-toggle-entity-row.ts]: No need to check for `stateObj` here, if it didn't exist, you would already have crashed on line 46 😉

Comment (balloob) [reply src/panels/lovelace/entity-rows/hui-toggle-entity-row.ts]: I think that you can just put `computeStateDisplay` directly in to the template above.

File: src/panels/lovelace/entity-rows/hui-toggle-entity-row.ts (hunk 1 above)
Comment (balloob) [reply src/panels/lovelace/entity-rows/hui-toggle-entity-row.ts]: Ooh, `hui-generic-entity-row` will render a yello block if the entity doesn't exist, we just need to make sure we don't crash on line 46.

File: src/panels/lovelace/common/create-row-element.js (hunk 3 above)
Comment (iantrich) [reply src/panels/lovelace/common/create-row-element.js]: Just remember that you asked for this: https://github.com/home-assistant/home-assistant-polymer/pull/1941
I started small with only all of `src` 😄

File: src/panels/lovelace/entity-rows/hui-toggle-entity-row.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  public setConfig(config: EntityConfig): void {
+    if (!config || !config.entity) {
+      throw new Error("Invalid Configuration: 'entity' required");
+    }
+
+    this._config = config;
+  }
+
+  protected render(): TemplateResult {
+    if (!this._config || !this.hass || !this.hass.states[this._config.entity]) {
```
Comment (balloob) [inline src/panels/lovelace/entity-rows/hui-toggle-entity-row.ts]: This is not good. Right now we won't render anything if the entity doesn't exist.

The old behavior was that if the entity doesn't exist, we still render `<hui-generic-entity-row>` which would then go ahead and render a yellow warning that the entity didn't exist.

**Review by balloob — APPROVED** [review]

**Pull request merged**